Solutions/Branwen/GUI.cs

```diff
@@ -0,0 +1,166 @@
+                Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild<Sheets>(new Sheets());
+
+                //Loop through all Directories and make a new sheet and put that shit in there
+                for (int i = 0; i < topLevelDirectories.Length; i++)
+                {
+                    WorksheetPart worksheetPart = workbookpart.AddNewPart<WorksheetPart>();
+                    worksheetPart.Worksheet = new Worksheet(new SheetData());
+                    SheetData sheetData = worksheetPart.Worksheet.GetFirstChild<SheetData>();
+                    Sheet sheet = new Sheet();
+                    sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
+                    sheet.Name = topLevelDirectories[i].Name;
+                    sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
+                    sheets.Append(sheet);
+                    WriteDirectoryToWorksheet(RunInventory(topLevelDirectories[i]), sheetData);
+                }
+                spreadsheetDocument.Close();
+                MessageBox.Show("DONE! Files Inventoried:  " + fileCount);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Failed to Inventory:" + Environment.NewLine + ex.Message);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+        }
+
+        /// <summary>
+        /// Recursively finds all Files in the parent directory
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <returns></returns>
+        private List<FileInfo> RunInventory(DirectoryInfo parent)
+        {
+            List<FileInfo> toReturn = new List<FileInfo>();
+            foreach (DirectoryInfo directory in parent.GetDirectories())
+            {
+                toReturn.AddRange(RunInventory(directory));
+            }
+            toReturn.AddRange(parent.GetFiles());
+            return toReturn;
+        }
+
+        /// <summary>
+        /// Writes all files in the Directory to the Workbook
+        /// </summary>
+        /// <param name="files"></param>
+        /// <param name="writer"></param>
+        private void WriteDirectoryToWorksheet(List<FileInfo> files, SheetData sheetData)
+        {
+            //Write Worksheet Header
+            WriteHeader(sheetData);
+
+            for (int i = 0; i < files.Count; i++)
+            {
+                Row row = new Row();
+                Cell cell = new Cell();
+                
+                //File Name
+                cell.CellValue = new CellValue(Path.GetFileNameWithoutExtension(files[i].Name));
+                cell.DataType = CellValues.String;
+                row.Append(cell);
+
+                //File Extension
+                cell = new Cell();

```

Comment:
I don't think you need to reinit the object each time, just take this line out and allow for the next two lines to simply change the values of the existing object.